client/me/account/index.jsx

```diff
@@ -210,6 +210,13 @@ module.exports = React.createClass( {
 		} );
 	},
 
+	onSiteSelect( siteSlug ) {
+		let selectedSite = sites.getSite( siteSlug );
+		if ( sites.getSite( siteSlug ) ) {
+			this.props.userSettings.updateSetting( 'primary_site_ID', selectedSite.ID );
```

Comment:
This is specifically for the `userSettings` module which doesn't have flux. The way we handle data for `user` and `userSettings` could definitely use some work.